openlibrary/components/BarcodeScanner.vue

```diff
@@ -0,0 +1,322 @@
+                Advanced
+        </summary>
+        <div class="barcodescanner__controls">
+          <button class="barcodescanner__read-isbn glass-button">
+            Read Text ISBN
+            <br>
+            <small>For books that print the ISBN without a barcode</small>
+          </button>
+        </div>
+      </details>
+      <div class="barcodescanner__result-strip">
+        <LazyBookCard v-if="lastISBN" :isbn="lastISBN" :state="state"/>
+        <div class="empty">
+          Point your camera at a barcode! 📷
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import OLBarcodeScanner from './BarcodeScanner/components/OLBarcodeScanner.vue';
+import LazyBookCard from './BarcodeScanner/components/LazyBookCard.vue';
+import Promise from 'promise-polyfill';
+import CONFIGS from './configs';
+
+export default {
+    components: { OLBarcodeScanner, LazyBookCard },
+    data() {
+        return {
+            lastISBN: null,
+            state: {},
+        }
+    },
+    methods: {
+        updateLastISBN(variable) {
+            this.lastISBN = variable;
+            this.fromISBN(variable);
+        },
+
+        updateState(newState) {
```

Comment:
Haha these helper methods existed in LazyBookCard.js because I wasn't using Vue 😁 With Vue we can now take advantage of its reactivity! So in LazyBookCard, you can set the data to be all the individual fields:

```js
data() {
    return {
        coverSrc: null,
        link: null,
        ...
    }
}
```

And instead of calling `updateState`, we can just change the values directly inside LazyBookCard